### DISCUSSION TITLE
Is possible draw image like PNG or JPG?

### DISCUSSION THREAD

**Discussion by ghost** [alandefreitas/matplotplusplus#30] [Q&A]
I has algorithm make synthetic image every second, want display with graph plot. 

Is can have example not depend OPenCV?

**Comment by alandefreitas** [comment]
Hi. Have you tried the examples in [this category](https://github.com/alandefreitas/matplotplusplus#images)?
It shouldn't be a problem. 

The build script will look for the libraries to open [all these kinds of files](https://github.com/alandefreitas/matplotplusplus/blob/64bc04c2e5d4dcbac9ec02e86f6f78552d34b256/source/3rd_party/CMakeLists.txt#L70) on your computer. This is not embedded with the library though. We could embed it someday but it's not the way it is right now.

As for your other questions, if your image is very large, updating it very often might be inefficient because the [default backend](https://github.com/alandefreitas/matplotplusplus/blob/master/source/matplot/backend/gnuplot.h) is a Gnuplot process pipe. We are implementing an [OpenGL backend](https://github.com/alandefreitas/matplotplusplus/blob/master/source/matplot/backend/opengl_3.h), which is probably going to be more efficient for all tasks, but that's going to take a while. There is [an example](https://github.com/alandefreitas/matplotplusplus/blob/master/test/backends/main.cpp) using this experimental OpenGL backend, but it requires some [more dependencies](https://github.com/alandefreitas/matplotplusplus/blob/64bc04c2e5d4dcbac9ec02e86f6f78552d34b256/source/matplot/CMakeLists.txt#L128) like OpenGL itself.

As for your second question, the library does not depend on OpenCV at all. OpenCV is an [optional dependency](https://github.com/alandefreitas/matplotplusplus/blob/64bc04c2e5d4dcbac9ec02e86f6f78552d34b256/source/3rd_party/CMakeLists.txt#L135) for the CImg library, which is the library we use for handling images. [CImg is embeded](https://github.com/alandefreitas/matplotplusplus/blob/64bc04c2e5d4dcbac9ec02e86f6f78552d34b256/source/3rd_party/CMakeLists.txt#L41) with the library.
